runtime/router.go

```diff
@@ -250,30 +158,48 @@ func (router *HTTPRouter) Register(
 	)
 }
 
-func (router *HTTPRouter) handleNotFound(w http.ResponseWriter, r *http.Request) {
-	resFields := []zapcore.Field{
-		zap.Int(statusCodeZapName, 404),
-	}
-	writeLogs(router.gateway.Logger, logRequestFields(r), resFields)
+func (router *HTTPRouter) handleNotFound(
+	w http.ResponseWriter,
+	r *http.Request,
+) {
+	var startTime, finishTime time.Time
```

Comment:
would prefer if we just allocated an NewServerHTTPRequest() here and started it. Then we can finish the res associated with it.

Reply:
I don't think it's worth allocating the req/res objects here just to record the start and finish times.